ledger/src/blockstore.rs

```diff
@@ -10692,44 +10692,52 @@ pub mod tests {
     fn erasure_multiple_config() {
         solana_logger::setup();
         let slot = 1;
-        let parent = 0;
         let num_txs = 20;
-        let entry = make_large_tx_entry(num_txs);
-        let shreds = entries_to_test_shreds(
-            &[entry],
-            slot,
-            parent,
-            true,  // is_full_slot
-            0,     // version
-            false, // merkle_variant
-        );
-        assert!(shreds.len() > 1);
+        let entries = [make_large_tx_entry(num_txs)];
 
-        let ledger_path = get_tmp_ledger_path_auto_delete!();
-        let blockstore = Blockstore::open(ledger_path.path()).unwrap();
+        let version = solana_sdk::shred_version::version_from_hash(&entries[0].hash);
 
+        let shredder = Shredder::new(slot, 0, 0, version).unwrap();
         let reed_solomon_cache = ReedSolomonCache::default();
-        let coding1 = Shredder::generate_coding_shreds(
-            &shreds,
-            0, // next_code_index
+        let (data1, coding1) = shredder.entries_to_shreds(
+            &Keypair::new(),
+            &entries,
+            true, // is_last_in_slot
+            // chained_merkle_root
+            Some(Hash::new_from_array(rand::thread_rng().gen())),
+            0,    // next_shred_index
+            0,    // next_code_index
+            true, // merkle_variant
             &reed_solomon_cache,
+            &mut ProcessShredsStats::default(),
         );
-        let coding2 = Shredder::generate_coding_shreds(
-            &shreds,
-            1, // next_code_index
+        let (_data2, coding2) = shredder.entries_to_shreds(
+            &Keypair::new(),
+            &entries,
+            true, // is_last_in_slot
+            // chained_merkle_root
+            Some(Hash::new_from_array(rand::thread_rng().gen())),
+            0,    // next_shred_index
+            1,    // next_code_index
```

Comment:
Might be nice to add a comment explaining what this test is doing (Intentionally making the `ErasureMeta` mismatch via the code index so that we fail the config check and mark the slot duplicate)

Reply:
Done in https://github.com/anza-xyz/agave/commit/1ab952c1b508d6c8c08fc6fe7cd33c7d82e6440b